src/IterationTraits.php

```diff
@@ -225,16 +225,43 @@ public static function sortKeys(Iterator $iterator, $fn = null) {
      * Group all the the values into an array and return the result as a Sequence.
      *
      * @param Iterator $iterator
-     * @param callable  $fnToGroup
+     * @param callable $fnToGroup
+     * @param null $keys
      * @return Sequence
      */
-    public static function groupBy(Iterator $iterator, $fnToGroup) {
-        return self::wrapFunctionIntoSequenceOnDemand(function() use ($iterator, $fnToGroup) {
+    public static function groupBy(Iterator $iterator, $fnToGroup, $keys = null) {
```

Comment:
I created commit 3ccb649482925df11ebaf14a6876c5678fca1f4e on a separate branch. Let me know what you think and I can add it to this pull request.